src/pocket.ts

```diff
@@ -0,0 +1,239 @@
+import * as cg from './types';
+import * as util from './util';
+import { dragNewPiece } from './drag';
+import { setDropMode, cancelDropMode } from './drop';
+
+import { HeadlessState, State } from './state';
+import { Elements, PieceNode } from './types';
+import { predrop } from './predrop';
+
+// These types maybe belong to ./types.ts, but put here to avoid merge conflicts from upsteam
+type Position = 'top' | 'bottom';
+export type Pocket = Partial<Record<cg.Role, number>>;
+export type Pockets = Partial<Record<cg.Color, Pocket>>;
+export type PocketRoles = (color: cg.Color) => string[] | undefined; // type for functions that map a color to possible
+                                                                     // pieces that can be in pocket for that side
+
+export const eventsDragging = ['mousedown', 'touchmove'];
+export const eventsClicking = ['click'];
+
+/**
+ * Logically maybe belongs to fen.ts, but put here to avoid merge conflicts from upsteam
+ * Analogous to fen.ts->read(), but for pocket part of FEN
+ * TODO: See todo in fen.ts->read() as well. Not sure if pocket parsing belongs there unless return
+ *       type is extended to contain pocket state.
+ * */
```

Comment:
Yes, this should probably be merged with fenRead, but it'll probably involve merging Pieces and Pockets into one type and should be done after drop moves become truly first-class citizen.